src/gui/Src/Gui/CPUStack.cpp

```diff
@@ -406,7 +424,80 @@ QString CPUStack::paintContent(QPainter* painter, dsint rowBase, int rowOffset,
         painter->drawText(QRect(x + 4, y , w - 4 , h), Qt::AlignVCenter | Qt::AlignLeft, makeAddrText(wVa));
         return QString();
     }
-    return HexDump::paintContent(painter, rowBase, rowOffset, col, x, y, w, h);;
+    else if(col == 1) // paint stack data
+    {
+        // get call stack
+        DBGCALLSTACK callstack;
+        memset(&callstack, 0, sizeof(DBGCALLSTACK));
+        // TODO: need to cache the call stack data.
+        DbgFunctions()->GetCallStack(&callstack);

```

Comment:
Hm, possibly I cached this function some time ago so that means it won't be
a problem👍
On Thu, 28 Jul 2016 at 09:17, TPoDT|Mr. eXoDia <mr.exodia.tpodt@gmail.com>
wrote:

> Try loading some symbols for all dlls and then stepping deep in a call
> stack.
> On Thu, 28 Jul 2016 at 04:17, Torusrxxx <notifications@github.com> wrote:
>
>> In src/gui/Src/Gui/CPUStack.cpp
>> <https://github.com/x64dbg/x64dbg/pull/892#discussion_r72556100>:
>>
>> > @@ -406,7 +424,80 @@ QString CPUStack::paintContent(QPainter* painter, dsint rowBase, int rowOffset,
>> >          painter->drawText(QRect(x + 4, y , w - 4 , h), Qt::AlignVCenter | Qt::AlignLeft, makeAddrText(wVa));
>> >          return QString();
>> >      }
>> > -    return HexDump::paintContent(painter, rowBase, rowOffset, col, x, y, w, h);;
>> > +    else if(col == 1) // paint stack data
>> > +    {
>> > +        // get call stack
>> > +        DBGCALLSTACK callstack;
>> > +        memset(&callstack, 0, sizeof(DBGCALLSTACK));
>> > +        // TODO: need to cache the call stack data.
>> > +        DbgFunctions()->GetCallStack(&callstack);
>>
>> In fact I have not observed a significant performance loss. After all
>> there is only a few lines in the CPUStack that need to repaint.
>>
>> —
>> You are receiving this because you commented.
>>
>>
>> Reply to this email directly, view it on GitHub
>> <https://github.com/x64dbg/x64dbg/pull/892/files/3c3f4a1ae4e5d185af1943952c92a65be5ab4072#r72556100>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ACWCmVj-GLew0RsNUHf2YEDepQLAuoppks5qaBFKgaJpZM4JWRdJ>
>> .
>>
>